src/deadline/client/cli/_groups/asset_group.py

```diff
@@ -408,3 +451,123 @@ def update_manifest(manifest: str, new_or_modified_paths: List[tuple]) -> BaseAs
         manifest_file.write(local_base_asset_manifest.encode())
 
     return local_base_asset_manifest
+
+
+def compare_manifest(
+    reference_manifest: BaseAssetManifest, compare_manifest: BaseAssetManifest
+) -> List[(tuple)]:
+    """
+    Compares two manifests, reference_manifest acting as the base, and compare_manifest acting as manifest with changes.
+    Returns a list of FileStatus and BaseManifestPath
+
+    """
+    reference_dict = {
+        manifest_path.path: manifest_path for manifest_path in reference_manifest.paths
+    }
+    compare_dict = {manifest_path.path: manifest_path for manifest_path in compare_manifest.paths}
+
+    differences = []
+
+    # Find new files
+    for file_path, manifest_path in compare_dict.items():
+        if file_path not in reference_dict:
+            differences.append((FileStatus.NEW, manifest_path))
+        elif file_path in reference_dict:
+            if reference_dict[file_path].hash != manifest_path.hash:
+                differences.append((FileStatus.MODIFIED, manifest_path))
+            else:
+                differences.append((FileStatus.UNCHANGED, manifest_path))
+
+    # Find deleted files
+    for file_path, manifest_path in reference_dict.items():
+        if file_path not in compare_dict:
+            differences.append((FileStatus.DELETED, manifest_path))
+
+    return differences
+
+
+def pretty_print(file_status_list: List[(tuple)]):
+    """
+    Prints to command line a formatted file tree structure with corresponding file statuses
+    """
+
+    # ASCII characters for the tree structure
+    PIPE = "│"
+    HORIZONTAL = "──"
+    ELBOW = "└"
+    TEE = "├"
+    SPACE = "    "
+
+    # ANSI escape sequences for colors
+    COLORS = {
+        "MODIFIED": "\033[93m",  # yellow
+        "NEW": "\033[92m",  # green
+        "DELETED": "\033[91m",  # red
+        "UNCHANGED": "\033[90m",  # grey
+        "RESET": "\033[0m",  # base color
+        "DIRECTORY": "\033[90m",  # grey
+    }
+
+    # Tooltips:
+    TOOLTIPS = {
+        "NEW": " +",  # added files
+        "DELETED": " -",  # deleted files
+        "MODIFIED": " M",  # modified files
+        "UNCHANGED": "",  # unchanged files
+    }
```

Comment:
food for thought + not blocking - The keys here (MODIFIED/NEW/DELETED/UNCHANGED) have to match FileStatus Enum for the logic to work. Could we think of a way to improve that?

There is an approach to consider
- https://realpython.com/python-enum/#using-enumerations-in-if-and-match-statements

Reply:
This is definitely cleaner, and I've added that change for the tooltips, however since colors also contain non-filestatus colors, like directory, I don't think I can easily change that for the colors enum

Reply:
We could still use the enum for file status colors' keys.